v5: Icons!

mdo opened the pull request:
Replaces the other half of #29048.

- Drops the Extend docs and moves the icons page to the Components section so we can have our own set
- Adds docs for customizing and using Bootstrap Icons
- Renames the classes to `.bi`
- Optimizes (thanks @XhmikosR!)
- Replaces the carousel icons with new compact chevrons

Still todo:

- [ ] ~~Tabnav between grid and table view of icons?~~
- [x] Move icons outside of content folder
- [x] Stop inlining the SVGs; we have http/2 so it should be pretty fast anyway?
- [ ] Verify if we need to add `role="img"` or `focusable="false"`
- [ ] ~~Add a tooltip in each icon showing its title and/or class?~~
- [ ] Click to copy SVG code?
- [ ] Fix inline customize SVGs display

/cc @XhmikosR @twbs/css-review 

Preview: <https://deploy-preview-29092--twbs-bootstrap.netlify.com/docs/4.3/components/icons/>

mdo committed:
Drop the Extend docs and move their content

- Moves the extend/icons/ page content to components/icons/ for now (still needs work to rewrite)
- Moves the extend/approach/ page into the getting-started/ section

mdo committed:
Improve icons page

XhmikosR committed:
Optimize the new SVGs.

XhmikosR committed:
Emphasize

mdo committed:
more icons to help test things

mdo committed:
Rename icons to .bi for bootstrap icon

mdo committed:
Improve icons docs

mdo committed:
Bring in the actual SVGs (missing classnames), update the table, use a new simple shortcode

XhmikosR committed:
Optimize icons.

XhmikosR committed:
Automate icons docs.

mdo committed:
Update icons

- More new icons
- Add their classes
- Fix property order
- More customizing docs

mdo committed:
Add more icons

mdo committed:
Add a new grid view of the icons (hoping to add a tab nav between table and grid)

mdo committed:
Replace Carousel icons with new ones

XhmikosR committed:
Add SVGO config for inline SVGs usage.

TryHardDood commented:
`Add a tooltip in each icon showing its title and/or class?`

And maybe click to copy the icon's code?

patrickhlauke commented:
yes, when the icons are actually shown, they need to somehow be presented to assistive technology/screen reader users. the `<svg>` itself, or its parent container, need a `role="img"` and an appropriate name, possibly via `aria-label="..."` on the element with that role.

and yes, the raw `<svg>` elements will need `focusable="false"

patrickhlauke commented:
if planning to add a tooltip, the `<svg>` or container will need to be focusable though (so tooltip can be triggered via keyboard).

mdo committed:
Big icon update

- Move icons to the dist folder
- Add a icons-copy npm task to bring them into the ignored directory for serving in Hugo
- Switch from inlining SVG as HTML to using img element
- Remove the icon table and only use the grid

mdo committed:
Add more icons, add svgo npm script, change the display from dir reading to data yml

XhmikosR commented on site/data/icons.yml:
```diff
@@ -1,25 +1,97 @@
+- name: bootstrap-fill
+- name: reboot
+
+- name: x
```
I'd use the non-unicode `x`

XhmikosR committed:
Don't hardcode the path.

XhmikosR committed:
Add missing svgo devDependency.

XhmikosR committed:
Optimize inline SVGs.

mdo committed:
1px strokes across the board

XhmikosR commented on site/content/docs/4.3/components/icons.md:
```diff
@@ -11,21 +11,16 @@ toc: true
-  {{< listIcons.inline >}}
-  {{- $dirName := printf "site/static/docs/%s/dist/icons/" .Site.Params.docs_version -}}
-
-  {{- range (readDir $dirName) -}}
```
@mdo: is the only reason for this patch the fact that you want the Bootstrap icons shown first? This adds an extra step which we could avoid. I don't mind either way, just thinking out loud.

I could probably hook something up to change the order, but it might complicate things.

XhmikosR committed:
Merge branch 'master' into mdo-icons-update

XhmikosR committed:
Merge branch 'master' into mdo-icons-update

XhmikosR committed:
Merge branch 'master' into mdo-icons-update

XhmikosR committed:
Merge branch 'master' into mdo-icons-update
